### PR TITLE
minor changes to remove pandas warnings

### PR DESCRIPTION
**Pull request by gprichter**

### CODE REVIEW

**Comment by anitagraser** [comment]
Awesome, tested https://gke.mybinder.org/v2/gh/gprichter/movingpandas/master and the only remaining errors are:

/srv/conda/envs/notebook/lib/python3.6/site-packages/geopandas/base.py:76: UserWarning: Cannot generate spatial index: Missing package `rtree`.
  warn("Cannot generate spatial index: Missing package `rtree`.")

and

/home/jovyan/trajectory.py:72: UserWarning: Discarding nonzero nanoseconds in conversion
  if not self.get_start_time() < self.get_end_time():

File: trajectory.py [hunk 1]
```diff
@@ -248,15 +260,22 @@ def add_speed(self, overwrite=False):
...
+        if 't' in self.df.columns:
+            times = self.df.t
+        else:
+            times = self.df.reset_index().t
+        self.df = self.df.assign(delta_t=times.diff().values)
         try:
             self.df[SPEED_COL_NAME] = self.df.apply(self._compute_speed, axis=1)
         except ValueError as e:
             raise e
+        # ? what is that doing
```
Comment (anitagraser) [inline trajectory.py]: This sets the speed in the first row to the speed of the second row

File: trajectory.py [hunk 2]
```diff
@@ -248,15 +260,22 @@ def add_speed(self, overwrite=False):
...
         by total seconds.
         """
-        if SPEED_COL_NAME in self.df.keys() and not overwrite:
+        if SPEED_COL_NAME in self.df.columns and not overwrite:
             raise RuntimeError('Trajectory already has speed values! Use overwrite=True to overwrite exiting values.')
-        self.df['prev_pt'] = self.df.geometry.shift()
-        self.df['t'] = self.df.index
-        self.df['delta_t'] = self.df['t'].diff()
+        self._update_prev_pt()
+        # NOTE: ? do we need the 't' again
```
Comment (anitagraser) [inline trajectory.py]: Probably not

File: trajectory.py [hunk 3]
```diff
@@ -187,14 +187,25 @@ def _compute_distance(self, row):
...
     def get_length(self):
         """Return float of length of Trajectory object.
 
         This is calculated with the measurement unit of the CRS used, except
         when using WGS 84 when it is calculated in metres.
         """
-        self.df['prev_pt'] = self.df.geometry.shift()
-        self.df['dist_to_prev'] = self.df.apply(self._compute_distance, axis=1)
+        self._update_prev_pt()
+        # NOTE: we could also make this column temporary if not needed again...
```
Comment (anitagraser) [inline trajectory.py]: Could be a good idea to avoid inconsistencies later on

File: trajectory_sampler.py [hunk 4]
```diff
@@ -67,6 +69,8 @@ def _is_sampling_possible(self, past_timedelta, future_timedelta, min_meters_per
         return True 
         
     def _get_actually_available_times(self, start_time, past_time, future_time):
+        # NOTE: this is a bit crufty, as everything about the times 
+        #   should be available in the index
```
Comment (anitagraser) [inline trajectory_sampler.py]: Not sure what to make of this comment.

File: trajectory.py (hunk 1 above)
Comment (gprichter) [reply trajectory.py]: thanks for the explanation, maybe as a comment in source?

File: trajectory_sampler.py (hunk 4 above)
Comment (anitagraser) [reply trajectory_sampler.py]: The logic of TrajectorySampler is explained in https://anitagraser.com/2018/12/15/movement-data-in-gis-18-creating-evaluation-data-for-trajectory-predictions/

**Pull request merged**